Configurations/unix-Makefile.tmpl

```diff
@@ -531,10 +537,14 @@ build_all_generated: $(GENERATED_MANDATORY) $(GENERATED) build_docs
 	@echo "         then make will fail..."
 	@ : {- output_on() if $disabled{makedepend}; "" -}
 
-all: build_sw build_docs
+##@ Help
+.PHONY: help
+help: ## Show this help screen
+	@$(PERL) $(SRCDIR)/util/help.pl $(MAKEFILE_LIST)
```

Comment:
`MAKEFILE_LIST` is a GNUism, not something universal, unfirtunately.  According to [FreeBSD make(1)](https://man.freebsd.org/cgi/man.cgi?make(1)), the corresponding on FreeBSD (and presumably Net- and OpenBSD), the corresponding variable is `.MAKE.MAKEFILES`.

Looking back into older (and I mean *older*) manuals, there's also the potential of a `make` implementation with no variable like this at all.

This could be fine, really...........  if it wasn't for the `while (<>)` construct in that perl script, which will try to read from stdin if there are no arguments at all!

Reply:
Or just change `$(MAKEFILE_LIST)` to `$(SRCDIR)/Makefile`

Reply:
> Or just change `$(MAKEFILE_LIST)` to `$(SRCDIR)/Makefile`

You mean `$(BLDDIR)/Makefile`...  Yup, that would be a way to do it

Reply:
Done.